src/app/_components/PageHeader.tsx

```diff
@@ -8,25 +8,84 @@ import {
   DescriptionText,
 } from '@/components/DescriptionText'
 import { Heading } from '@/components/Heading'
-import { Meta } from '@/components/Meta'
+import { Meta, type MetaDataType } from '@/components/Meta'
+import { NextStaticImage } from '@/components/NextStaticImage'
 import { SectionDivider } from '@/components/SectionDivider'
 
 import { type CTAProps } from '@/types/sharedProps/ctaType'
-import { type ImageProps } from '@/types/sharedProps/imageType'
+import {
+  type StaticImageProps,
+  type ImageProps,
+} from '@/types/sharedProps/imageType'
 
 type TitleProps = {
   children: string
 }
 
-type PageHeaderProps = {
+type SharedProps = {
   title: TitleProps['children']
   description?: DescriptionTextType
-  cta?: CTAProps
-  secondaryCta?: CTAProps
-  image?: ImageProps
-  metaData?: Array<string | null | undefined>
+  cta: CTAProps | [CTAProps, CTAProps?]
```

Comment:
In the case of the array - How come the second `CTAProps` is optional?

Reply:
Also why do we need a separate `SharedProps` type? Instead of just a `PageHeaderProps`?

```
type PageHeaderProps = {
  title: TitleProps['children']
  cta: CTAProps | [CTAProps, CTAProps]
  description?: DescriptionTextType
  metaData?: MetaDataType
  isFeatured?: boolean
  image:
    | ({ type: 'static' } & StaticImageProps)
    | ({ type: 'dynamic' } & DynamicImageProps)
}
```

Reply:
Yes good catch, the second `CTAProps` in the array doesn't have to be optional.

Reply:
And also yes, we can group all the types in `PageHeaderProps`

Reply:
✅